frontend/admin/src/js/components/Layout.tsx

```diff
@@ -49,8 +53,17 @@ const Layout = () => {
   const isSmallScreen = useIsSmallScreen();
   const [isMenuOpen, setMenuOpen] = React.useState(!isSmallScreen);
 
+  // retain state of admin sidemenu in local storage
+  const menuStorageKey = "digitaltalent-menustate";
+  const menuPreference = getFromLocalStorage(menuStorageKey, true);
+  if (isMenuOpen !== menuPreference) {
+    // prevent infinite re-render with a conditional execution
+    setMenuOpen(menuPreference);
+  }
+
```

Comment:
Sorry, I'm not exactly sure what effect this change makes?

Reply:
Yeah, so there is a few reasons I suggested this:

1. We can initialize the state from preference rather than checking it later and setting it after render
2. Removes need for the conditional that runs on ever render, it now only happens on mount when state is initialized
3. Updated state on render right after initialization can lead to state flashes on slower hardware (unlikely but possible)

By passing either `menuPreference || !isSmallScreen`, we are initializing the state directly with the comparison avoiding any possible flashes of state changes on render and we no longer have to do the conditional check on every render.

Reply:
yeah, that would be better to condense things into one go rather than render -> re-render
feels smoother than way too, thanks for the detailed explanation

Reply:
tackled in https://github.com/GCTC-NTGC/gc-digital-talent/pull/5351/commits/e6ca6a9378fa4808b4244931ee21ef77eb364358

`menuPreference || !isSmallScreen`
didn't really work well as if preference was open (true) it never checks the after the `| |` so I had to adjust things a bit